### PR TITLE
Handle file/pipe stdout and add fallback on Windows

### LINKED ISSUE #4422: Console output does not work with Git Bash
Git Bash is a bash emulator that comes with Git for Windows.

It's useful for example because it supports displaying unicode characters.

It works with upstream teeworlds but not with the ddnet tools (e.g. `map_convert` prints output in cmd and powershell but not in Git Bash).

### PR DESCRIPTION
**Pull request by Robyt3**
The `CWindowsConsoleLogger` uses `WriteConsoleW` which only works when the standard output handle is a console (`GetFileType` returns `FILE_TYPE_CHAR`).

When redirecting output into a file, `GetFileType` returns `FILE_TYPE_DISK`. For this case a `CWindowsFileLogger` is added that uses `WriteFile` instead. This works on raw bytes, so the UTF-8 encoded lines are written directly.

When pipeing output or when using custom terminals like Git Bash, `GetFileType` returns `FILE_TYPE_PIPE`. I can't find any documentation how to correctly interact with a pipe stdout handle. So in this case, `WriteFile` is also used, because it works for me with cmd, PowerShell and Git Bash.

Lastly, if the output device is unknown, the default `CLoggerAsync` without ANSI colors is used as fallback.

This fixes output not working in Git Bash. Closes #4422. It also fixes output redirected with Git Bash containing additional line breaks.

For consistency, the `CWindowsConsoleLogger` now writes windows line endings also.

## Checklist

- [X] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test if it works standalone, system.c especially
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

**Comment by Robyt3** [comment]
Although, encoding appears to be broken when using unicode and pipeing output in PowerShell, e.g. with `./map_convert_07 "öüä" | more`.

File: src/base/log.cpp [hunk 1]
```diff
@@ -336,11 +342,26 @@ class CWindowsConsoleLogger : public ILogger
...
+
+public:
+	CWindowsFileLogger(HANDLE pFile) :
+		m_pFile(pFile)
+	{
+	}
+	void Log(const CLogMessage *pMessage) override
+	{
+		WriteFile(m_pFile, pMessage->m_aLine, pMessage->m_LineLength, NULL, NULL);
+		WriteFile(m_pFile, "\r\n", 2, NULL, NULL);
```
Comment (hrxi) [inline src/base/log.cpp:362]: This is not thread-safe, the newline might be printed a lot later. I think one can make it thread-safe by allocating a buffer for the whole message plus newline.

**Comment by Chairn** [comment]
As a user of git bash on Windows, thank you :heart:

**Comment by heinrich5991** [comment]
> Although, encoding appears to be broken when using unicode and pipeing output in cmd or PowerShell, e.g. with `./map_convert_07 "öüä" | more`.

Do you know how one could fix that?

**Pull request merged**